File: Recursion/MaxMinInArray/MaxMin.py
""" 
Problem : Find maximum and minimum element in array
"""

def maxElement(arr,index) : 
    if index >= len(arr) - 1:
        return arr[index]
    
    return max(maxElement(arr,index+1),arr[index])
    


def minElement(arr,index) : 
    if index >= len(arr) - 1:
        return arr[index]
    
    return min(minElement(arr,index+1),arr[index])
    


arr = [2,45,78,0,-9,52,-987,1006]

print("Maximum Element : ",maxElement(arr,0))
print("Minimum Element : ",minElement(arr,0))

File: Recursion/MaxMinInArray/MaxMin.java
/*
Problem : Find maximum and minimum element in array
*/


public class MaxMin{

    public static int maxElement(int[] arr,int index){
        if(index >= arr.length - 1){
            return arr[index];
        }

        return Math.max(maxElement(arr, index+1), arr[index]);
    }

    public static int minElement(int[] arr,int index){
        if(index >= arr.length - 1){
            return arr[index];
        }

        return Math.min(minElement(arr, index+1), arr[index]);
    }
    public static void main(String[] args){

        int[] arr = {2,45,78,0,-9,52,-987,1006};

        System.out.println("Maximum Element : " + maxElement(arr, 0));
        System.out.println("Minimum Element : " + minElement(arr, 0));

    }
}